src/classes/RD2_OpportunityService_TEST.cls

```diff
@@ -924,6 +924,120 @@ private with sharing class RD2_OpportunityService_TEST {
         assertPaymentIsMatchingOpp(oppById);
     }
 
+    /***
+    * @description Verifies if Account is not getting removed in opportunity
+    * when recurring donation record with only contact is updated.
+    */
+    @isTest
+    private static void shouldNotRemoveAccountInOppWhenRDWithNoAccountIsModified() {
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        setUpConfiguration();
+
+
+        npe03__Recurring_Donation__c rd = createRecurringDonation();
+
+        //create opp from rd with contact only
+        Opportunity opp = getOpportunityBuilderWithNoAccount(rd)
```

Comment:
The method is not required since it is used only once. The Opp builder details can be specified inline.